scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -120,34 +115,22 @@ public void initialize() {
   }
 
   /**
-   * Set up a timer to display the images at the specified frame rate from the combobox.
+   * Set up a timer to display the images at the specified speed from the combobox.
    */
   private void startNewAnimationTimer() {
 
-    timer = new Timer(true);
-    TimerTask timerTask = new TimerTask() {
-      @Override
-      public void run() {
-        setNextImageFrameToImageOverlay();
-      }
-    };
-    timer.scheduleAtFixedRate(timerTask, 1L, period);
+    animation = new Timeline();
+    animation.setCycleCount(-1); // loop animation
+    animation.getKeyFrames().add(new KeyFrame(Duration.millis(period), e -> {
+      // set image frame to image overlay
+      imageOverlay.setImageFrame(imageFrames.get(frameIndex));
+      // update to the next frame
+      frameIndex = (frameIndex + 1) % imageFrames.size();
```

Comment:
this won't work because it would just count up infinitely: going beyond the total size of the array which is `240`. The modulo here keeps the logic concise and neat, otherwise we'd have to do as I had before which adds more codelines: 

```
    frameIndex++;	
    // reset index once all files have been loaded	
    if (frameIndex == imageFrames.size())	
      frameIndex = 0;
```

Reply:
True, but at the cost of 2 extra lines of code and a comment, that is much more easy to comprehend